### PR TITLE
fix: flmodel load & save model

### PR DESCRIPTION
**Pull request by zhangxingmeng**

### CODE REVIEW

**Comment by zhangxingmeng** [comment]
I have read the CLA Document and I hereby sign the CLA

File: secretflow/ml/nn/fl/fl_model.py [hunk 1]
```diff
@@ -619,18 +623,24 @@ def save_model(self, model_path: Union[str, Dict[PYU, str]], is_test=False):
...
             if is_test:
-                model_path_test = os.path.join(
-                    model_path[device], device.__str__().strip("_")
+                device_model_path = os.path.join(
+                    device_model_path, device.__str__().strip("_")
                 )
-                res.append(worker.save_model(model_path_test))
-            else:
-                res.append(worker.save_model(model_path[device]))
+            if not os.path.exists(device_model_path):
```
Comment (zhouaihui) [inline secretflow/ml/nn/fl/fl_model.py:632]: `if not os.path.exists(device_model_path)` creates dirs on driver node, this behavior is expected to done on device node, so worker creates dir locally is necessary but not here.

**Review by zhouaihui — CHANGES_REQUESTED** [review]

File: secretflow/ml/nn/fl/fl_model.py [hunk 2]
```diff
@@ -619,18 +623,22 @@ def save_model(self, model_path: Union[str, Dict[PYU, str]], is_test=False):
         res = []
         for device, worker in self._workers.items():
             assert device in model_path, f'Should provide a path for device {device}.'
-            if not os.path.exists(model_path[device]):
-                os.makedirs(model_path[device])
+            device_model_path, device_model_name = model_path[device].rsplit("/", 1)
```
Comment (zhouaihui) [inline secretflow/ml/nn/fl/fl_model.py:626]: For input '/a/b' and '/a/b/', we got different output path '/a'，'/a/b'. But '/a/b' and '/a/b/' could mean the same dir.

Comment (zhangxingmeng) [reply secretflow/ml/nn/fl/fl_model.py:627]: model_path must be a/b/c， a/b/c/will be banned

**Comment by miluoalbert** [comment]
#196

**Comment by 6fj** [comment]
fix #196

**Review by zhouaihui — APPROVED** [review]
LGTM

**Pull request merged**